src/blueprint/operators/call.rs

```diff
@@ -0,0 +1,168 @@
+use std::collections::hash_map::Iter;
+
+use crate::blueprint::*;
+use crate::config;
+use crate::config::{Field, GraphQLOperationType, KeyValues};
+use crate::lambda::Expression;
+use crate::mustache::{Mustache, Segment};
+use crate::try_fold::TryFold;
+use crate::valid::{Valid, Validator};
+
+fn find_value<'a>(args: &'a Iter<'a, String, String>, key: &'a String) -> Option<&'a String> {
+    args.clone()
+        .find_map(|(k, value)| if k == key { Some(value) } else { None })
+}
+
+pub fn update_call(
+    operation_type: &GraphQLOperationType,
+) -> TryFold<'_, (&ConfigModule, &Field, &config::Type, &str), FieldDefinition, String> {
+    TryFold::<(&ConfigModule, &Field, &config::Type, &str), FieldDefinition, String>::new(
+        move |(config, field, _, _), b_field| {
+            let Some(call) = &field.call else {
+                return Valid::succeed(b_field);
+            };
+
+            compile_call(field, config, call, operation_type)
+                .map(|resolver| b_field.resolver(Some(resolver)))
+        },
+    )
+}
+
+pub fn compile_call(
```

Comment:
I think this implementation can be massively simplified. We need to invent a new expression inside `Context`

```rust
pub enum Context {
    Value,
    Path(Vec<String>),
    
    // A new operator
    CallWith {
        args: HashMap<String, Vec<String>>,
        expression: Box<Expression>
    } 
}
```

Whenever we see a `call` config, we should use it's `args` field to modify the current args in context and then call the expression with the new context. That way we won't need to make changes here, when we add a new IO type.

Reply:
Mentioned on Discord but updating it here also: tried to implement this but did not succeed. I do have a WIP on [feat/#846/create-call-operator-via-resolver-with-arg](https://github.com/ologbonowiwi/tailcall/tree/feat/%23846/create-call-operator-via-resolver-with-arg), though.

Reply:
Adding a todo here — https://github.com/tailcallhq/tailcall/issues/1285